hyperspy/signal.py

```diff
@@ -3885,9 +3885,12 @@ def _ma_workaround(self, s, function, axes, ar_axes, out):
         data = self.data.reshape(new_shape).squeeze()
 
         if out:
-            function(data, axis=ar_axes[0], out=out.data)
+            if np_out:
+                function(data, axis=ar_axes[0], out=out.data,)
+            else:
+                out.data[:] = function(data, axis=ar_axes[0],)
```

Comment:
This and the identical line in `_apply_function_on_data_and_remove_axis` enable things like that:
```python
In [2]: s = hs.signals.Spectrum(np.zeros((5,100)))

In [3]: ss = s.sum()

In [4]: ss.data.shape
Out[4]: (100,)

In [5]: ss.data[:5]
Out[5]: array([ 0.,  0.,  0.,  0.,  0.])

In [6]: s.data[:] = 1

In [7]: s.sum(axis=(0,1), out=ss)

In [8]: ss.data.shape
Out[8]: (100,)

In [9]: ss.data[:5]
Out[9]: array([ 500.,  500.,  500.,  500.,  500.])

In [10]: sum_real = s.sum(axis=(0,1))

In [11]: sum_real.data
Out[11]: array([ 500.])
```